demo/ssn-input.html

```diff
@@ -34,63 +33,107 @@
+        need to do this, unless you are trying to vend an element that uses
+        paper-input or paper-input-container _and_ works in both Polymer 1.0 or 2.0
+      -->
+       <span id="template-placeholder"></span>
     </div>
-
   </template>
-</dom-module>
 
-<script>
-  Polymer({
-    is: 'ssn-input',
+  <template id="v0">
+    <input is="iron-input" bind-value="{{_ssn1}}" maxlength="3" size="3" aria-label="First 3 digits of social security number">
+    -
+    <input is="iron-input"  bind-value="{{_ssn2}}" maxlength="2" size="2" aria-label="Middle 2 digits of social security number">
+    -
+    <input is="iron-input"  bind-value="{{_ssn3}}" maxlength="4" size="4" aria-label="Last 4 digits of social security number">
+  </template>
 
-    behaviors: [
-      Polymer.IronValidatableBehavior
-    ],
+  <template id="v1">
+    <iron-input bind-value="{{_ssn1}}" aria-label="First 3 digits of social security number">
+      <input maxlength="3" size="3">
+    </iron-input>
+    -
+    <iron-input bind-value="{{_ssn2}}" aria-label="Middle 2 digits of social security number">
+      <input maxlength="2" size="2">
+    </iron-input>
+    -
+    <iron-input bind-value="{{_ssn3}}" aria-label="Last 4 digits of social security number">
+      <input maxlength="4" size="4">
+    </iron-input>
+  </template>
 
-    properties: {
-      value: {
-        notify: true,
-        type: String
+  <script>
+    Polymer({
+      is: 'ssn-input',
+
+      behaviors: [
+        Polymer.IronValidatableBehavior
+      ],
+
+      properties: {
+        value: {
+          notify: true,
+          type: String
+        },
+
+        _ssn1: {
+          type: String,
+          value: ''
+        },
+
+        _ssn2: {
+          type: String,
+          value: ''
+        },
+
+        _ssn3: {
+          type: String,
+          value: ''
+        },
+
+        validator: {
+          type: String,
+          value: 'ssn-validator'
+        }
       },
 
-      _ssn1: {
-        type: String
-      },
+      observers: [
+        '_computeValue(_ssn1,_ssn2,_ssn3)'
+      ],
 
-      _ssn2: {
-        type: String
+      _computeValue: function(ssn1, ssn2, ssn3) {
+        this.value = ssn1.trim() + '-' + ssn2.trim() + '-' + ssn3.trim();
       },
 
-      _ssn3: {
-        type: String
+      // TODO: remove this when 2.0 adds beforeRegister back to legacy
+      registered: function() {
+        if (Polymer.Element) {
+          this._beforeRegister();
+        }
       },
 
-      validator: {
-        type: String,
-        value: 'ssn-validator'
-      }
-    },
-
-    observers: [
-      '_computeValue(_ssn1,_ssn2,_ssn3)'
-    ],
+      beforeRegister: function() {
+        this._beforeRegister();
+      },
 
-    _computeValue: function(ssn1, ssn2, ssn3) {
-      this.value = ssn1.trim() + '-' + ssn2.trim() + '-' + ssn3.trim();
-    }
-  });
-</script>
+      _beforeRegister: function() {
```

Comment:
Sorry, Kevin had briefly removed `beforeRegister` from 2.0, but I think it's back now. Will fix.